include/boost/geometry/algorithms/closest_points.hpp

```diff
@@ -0,0 +1,364 @@
+// Boost.Geometry
+
+// Copyright (c) 2019, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_ALGORITHMS_CLOSEST_POINTS_HPP
+#define BOOST_GEOMETRY_ALGORITHMS_CLOSEST_POINTS_HPP
+
+#include <boost/geometry/algorithms/clear.hpp>
+#include <boost/geometry/algorithms/distance.hpp>
+#include <boost/geometry/algorithms/detail/closest_points/default_strategies.hpp>
+#include <boost/geometry/algorithms/detail/convert_point_to_point.hpp>
+#include <boost/geometry/algorithms/not_implemented.hpp>
+#include <boost/geometry/core/closure.hpp>
+#include <boost/geometry/core/exception.hpp>
+#include <boost/geometry/core/point_type.hpp>
+#include <boost/geometry/core/tag.hpp>
+#include <boost/geometry/core/tags.hpp>
+#include <boost/geometry/strategies/default_strategy.hpp>
+#include <boost/geometry/strategies/densify.hpp>
+#include <boost/geometry/util/condition.hpp>
+#include <boost/geometry/util/range.hpp>
+
+#include <boost/range/size.hpp>
+#include <boost/range/value_type.hpp>
+
+#include <boost/throw_exception.hpp>
+
+
+
+namespace boost { namespace geometry
+{
+
+
+#ifndef DOXYGEN_NO_DISPATCH
+namespace dispatch
+{
+
+template
+<
+    typename Geometry1, typename Geometry2, typename Strategy
+>
+struct closest_points
+{
+    template <typename Segment>
+    static inline void apply(
+        Geometry1 const& g1,
+        Geometry2 const& g2,
+        Segment& shortest_seg,
+        Strategy const& strategy)
+    {
+        typename strategy::distance::services::return_type
+        <
+            Strategy,
+            Geometry1,
+            typename point_type<Geometry2>::type
+        >::type result = distance <
+                                      Geometry1, Geometry2, Strategy
+                                  >::apply(g1, g2, strategy);
```

Comment:
The indentation of <...> blocks is not correct in this section I think.

Reply:
So `closest_points` strategies are basically `distance` strategies but instead of the distance value 2 points are returned. And I'm saying 2 points because when we move to other dimensions then probably more coordinates will be returned here. Right?

Reply:
Yes, in particular, `closest_points` strategies return the coordinates of the two points and the distance (actually only the comparable distance is required since it is used only for comparison of  two `closest_points` results). As it is now the code assumes 2 dimensions.